dev/NavigationView/NavigationView.cpp

```diff
@@ -1326,13 +1359,105 @@ void NavigationView::ChangeSelection(const winrt::IInspectable& prevItem, const
     }
 }
 
+void NavigationView::UpdateIsChildSelected(winrt::IInspectable const& prevItem, winrt::IInspectable const& nextItem)
+{
+    // REMOVE CHECK ONCE TOP NAV FUNCTIONALITY IS IMPLEMENTED
+    if (IsTopNavigationView() || RootNodes().Size() == 0)
+    {
+        return;
+    }
+
+    auto lv = GetActiveListView();
+    auto viewModel = winrt::get_self<NavigationViewList>(lv)->ListViewModel();
+
+    if (lv && prevItem && !IsSettingsItem(prevItem))
+    {
+        winrt::TreeViewNode prevItemNode{ nullptr };
+
+        if (auto container = lv.ContainerFromItem(prevItem))
+        {
+            prevItemNode = NodeFromContainer(container);
+        }
+        else
+        {
+            prevItemNode = NodeFromPreviouslySelectedItem(prevItem);
+        }
+
+        MUX_ASSERT(prevItemNode);
+
+        if (prevItemNode && prevItemNode.Parent())
+        {
+            auto nodeParent = prevItemNode.Parent();
+            viewModel->UpdateSelection(nodeParent, TreeNodeSelectionState::UnSelected);
+            viewModel->NotifyContainerOfSelectionChange(nodeParent, TreeNodeSelectionState::UnSelected);
+        }
+    }
+
+    if (lv && nextItem && !IsSettingsItem(nextItem))
+    {
+        // The next item being selected must be in the listview
+        if (auto container = lv.ContainerFromItem(nextItem))
+        {
+            auto nextItemNode = NodeFromContainer(container);
+
+            MUX_ASSERT(nextItemNode);
+
+            if (nextItemNode && nextItemNode.Parent())
+            {
+                auto nodeParent = nextItemNode.Parent();
+                viewModel->UpdateSelection(nodeParent, TreeNodeSelectionState::PartialSelected);
+                viewModel->NotifyContainerOfSelectionChange(nodeParent, TreeNodeSelectionState::PartialSelected);
+            }
+        }
+    }
+}
+
+// This search function is utilizing the assumption that the item we are searching for in the node tree is in the process of being unselected.
+// This means that it's ancestors are still in the TreeNodeSelectionState::PartialSelected state.
+winrt::TreeViewNode NavigationView::NodeFromPreviouslySelectedItem(winrt::IInspectable const& item)
+{
+    auto nodeList = RootNodes();
+
+    bool updatedNodeList = true;
```

Comment:
>updatedNodeList [](start = 9, length = 15)

It has the same logic even if we remove updatedNodeList

Reply:
Yep, this doesn't affect the loop logic, I added this boolean as a preventative measure against an infinite loop in case, for some unforeseen reason, the item we are looking for is not found.